src/compiler/checker.ts

```diff
@@ -28128,19 +28175,45 @@ namespace ts {
             propertyName: __String,
             type: Type): boolean {
 
+            // Short-circuiting for improved performance.

```

Comment:
Copied this from the existing `isValidPropertyAccessWithType` function.
It seemed to be there for performance reasons, doesn't look like it would change behavior, so I added it assuming someone had a good reason to write it in the first place.